### ISSUE TITLE
`misc-const-correctness.WarnPointersAsValues` not detected with non-const member call

### ISSUE THREAD

**Issue by firewave** [llvm/llvm-project#64955]
```cpp
class T
{
public:
    void f(); // make function const for warning to appear
};

void f()
{
    T* /*const*/ t = new T();
    t->f();
}
```
https://godbolt.org/z/TW3r5Y1d8

The pointer can be `const`. As this is just about the variable and not the type it should not matter what kind of member is called on it.
The warning does triggers if you access a non-function member.
